open_event/helpers/data.py

```diff
@@ -166,6 +162,13 @@ def add_session_to_event(form, event_id, speaker_img_name, state="pending"):
                               user=login.current_user if login and login.current_user.is_authenticated else None)
         new_session.speakers.append(new_speaker)
         save_to_db(new_session, "Session saved")
+        speaker_img = ""
+        if speaker_img_file != "":
+            speaker_img = upload(speaker_img_file,
+                                 'events/%d/sessions/%d/avatar' % (int(event_id), int(new_session.id)))
```

Comment:
This is speaker's photo. So why not store in `events/id/speakers/id/photo` ? Is speaker not available ?
Also this method will fail if there are many speakers in a session.